### PR TITLE
Ch2 - Mining: Text and highlight border UX improvement

### PR DESCRIPTION
**Pull request by vinayaksh42**
fixes: #454

- Text fades in and fades out upon change.
- The highlight towards the end of the challenge ('tell me more' part) has a pulse effect in order to draw the attention of the user.

preview: https://saving-satoshi-git-fork-vinayaksh42-issue454-savingsatoshi.vercel.app/en/chapters/chapter-2/mining-1?dev=true

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
I don't see a pulse on the "tell me more" buttons. I like the fading text though.
https://github.com/saving-satoshi/saving-satoshi/assets/32662508/f50ce99a-6824-41f7-b074-a4ae017ca671

**Comment by vinayaksh42** [comment]
> I don't see a pulse on the "tell me more" buttons. I like the fading text though.
> https://github.com/saving-satoshi/saving-satoshi/assets/32662508/f50ce99a-6824-41f7-b074-a4ae017ca671

Oh sorry. I explained it wrongly, the pulse effect is on the border of mining components not on the button.

**Comment by ecurrencyhodler** [comment]
Oh then it's a tACK for me!

**Comment by GBKS** [comment]
Nice improvements already. Is it possible to also make the disappearing text fade out? And add a transition duration to the text elements in the miner UI. Particularly when going from disabled state to active state, the change is pretty abrupt. The tiny hash and the check/x need to instantly update, but all the other elements can have slight easing applied (150-250ms) to make it all just a little bit smoother.

**Review by shaavan — COMMENTED** [review]
I agree with @GBKS's suggestions. 

The text on the left feels easy on the eyes. Smooth and feels clean. While the right side felt a little rough on the eyes.
I could not put my finger on it, but Christoph summarised the reason well!

The mining challenge will look amazingly clean and smooth after making the text changes smooth in transition.

**Comment by vinayaksh42** [comment]
> I agree with @GBKS's suggestions.
> 
> The text on the left feels easy on the eyes. Smooth and feels clean. While the right side felt a little rough on the eyes. I could not put my finger on it, but Christoph summarised the reason well!
> 
> The mining challenge will look amazingly clean and smooth after making the text changes smooth in transition.

Yes, the start of the mining session looks a bit abrupt. Specially due to the text on the left side fading in and the mining session Starting way before the text appears.

File: content/lessons/chapter-2/mining-1.tsx [hunk 1]
```diff
@@ -90,38 +98,56 @@ export default function Mining1({ lang }) {
   }, [ramdomNonce])
 
   const transactionStep = () => {
-    setStep(5)
+    setShowText(false)
+    setTimeout(() => {
+      setStep(5)
+    }, 300)
```
Comment (afterburn) [inline content/lessons/chapter-2/mining-1.tsx:104]: Lets create a util function:
```ts
function sleep(timeout: number) {
  return new Promise(resolve => {
    const t = setTimeout(() => {
      resolve()
      clearTimeout(t)
    }, timeout)
  })
}
```

Then we can use it like this by making these function async:
```ts
const transactionStep = async () => {
  setShowText(false)
  setNonceHighlight(false)
  setHashPowerHighlight(false)
  setTransactionsConfirmedHighlight(true)
  await sleep(300)
  setStep(5)
})

File: ui/common/MiningStatistic.tsx [hunk 2]
```diff
@@ -19,57 +19,51 @@ export default function InfoBox({
...
-        <div
-          className={clsx('font-nunito text-[13px] font-bold', {
-            'text-black/25': bitcoin === 0,
-            'text-[#EDA081]': bitcoin !== 0,
-          })}
-        >
-          {bitcoinTitle}
+      </div>
+      <div className="relative h-[72px] w-1/2">
+        <div className="h-[72px] w-full rounded-[5px] border-2 border-transparent bg-black/15 p-4 px-[15px] py-2.5 font-space-mono ">
```
Comment (afterburn) [inline ui/common/MiningStatistic.tsx:46]: `72` is divisible by `4` so let's use h-18, if tailwind doesnt support this we can add this to our `tailwind.config.js` in the `extend` object.

File: ui/common/MiningStatisticsNonce.tsx [hunk 3]
```diff
@@ -35,62 +35,59 @@ export default function MiningStatisticNonce({
...
-        (blockFound % 4 === 0 && blockFound !== 0)) && (
-        <div className="order-last flex items-center justify-center">
-          <div className="flex  w-[76px] overflow-hidden font-space-mono text-[6px] !text-[#5DBC59]">
-            {formatHash(finalHash, 4, 4, 64, false, true)}
-          </div>
-          <Icon icon="greenCheck" className="h-8 w-8 py-2 text-[#5DBC59]" />
-        </div>
+      </div>
+      {highlight && (
+        <div className="absolute inset-0 mt-5 animate-pulse rounded-[5px] border-2 border-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]"></div>
```
Comment (afterburn) [inline ui/common/MiningStatisticsNonce.tsx:90]: Can we add the shadow to our `tailwind.config.js` file?

**Review by afterburn — CHANGES_REQUESTED** [review]
Thanks @vinayaksh42, left some comments.

File: ui/common/MiningStatistic.tsx (hunk 2 above)
Comment (vinayaksh42) [reply ui/common/MiningStatistic.tsx:46]: it goes from h-16 to h-20 to h-24. I will add h-18 (72px) as might get used in the future

**Review by afterburn — APPROVED** [review]

**Comment by afterburn** [comment]
@vinayaksh42 you have some conflicts, feel free to merge after resolving

**Review by shaavan — COMMENTED** [review]
The challenge looks a lot smoother now!

Just a few slight suggestions:

After the "Tell me more" button appears, the following text on the left-side fade-in smoothly, but the border highlight on the right feels slightly abrupt in its appearance and disappearance.


https://github.com/saving-satoshi/saving-satoshi/assets/85434418/9f582689-0f7d-4541-b9a5-85d571f40c2d

It will be a great addition to smooth the border's appearance too.

**Pull request merged**